cadc-util/src/test/java/ca/nrc/cadc/auth/SSOCookieManagerTest.java

```diff
@@ -79,30 +84,58 @@ public void roundTrip() throws Exception
     {
         final HttpPrincipal userPrincipal = new HttpPrincipal("CADCtest");
         SSOCookieManager cm = new SSOCookieManager();
-        DelegationToken cookieToken = cm.parse(cm.generate(userPrincipal));
+        DelegationToken cookieToken = cm.parse(cm.generate(userPrincipal, null));
         HttpPrincipal actualPrincipal = cookieToken.getUser();
-        
+
+        //Check principal
         assertEquals(userPrincipal, actualPrincipal);
     }
 
+    public String createCookieString() throws InvalidKeyException, IOException {
```

Comment:
This support methdo looks like it re-implements code in DelegationToken... It is going to every brittle.

I think a couple of tests like testRountTrip() -- testRoundTripMin() and testRoundTripAll() - with 1 identity (userid) and all supported identity types would be better.